### PR TITLE
Bug fix/touch resolution

### PR DESCRIPTION
**Pull request by fivetran-joemarkiewicz**
**Are you a current Fivetran customer?** 

Yes, original PR was created by @matthew-skinner

**What change(s) does this PR introduce?** 

This PR introduces updates to the is_touch resolution logic within the `zendesk__ticket_metrics` model. This logic now is updated to only calculate as a touch if the agent has a **public** comment attached to the ticket. As such, an additional field has been added to the intermediate `int_zendesk__comment_metrics` model which counts agent public comments.

**Does this PR introduce a breaking change?**

- [ ] Yes (please provide breaking change details below.)
- [X] No  (please provide explanation how the change is non breaking below.)
This PR only updates the logic for a final metric and adds a new field. No breaking changes.

**Is this PR in response to a previously created Issue**

- [X] Yes, Issue #32 
- [ ] No 

**How did you test the PR changes?** 

 
- [X] CircleCi  
- [ ] Other (please provide additional testing details below)

**Select which warehouse(s) were used to test the PR**
 
- [X] BigQuery
- [X] Redshift
- [X] Snowflake
- [X] Postgres
- [ ] Databricks
- [ ] Other (provide details below)

**Provide an emoji that best describes your current mood**
 
😈

### CODE REVIEW

File: models/zendesk__ticket_metrics.sql [hunk 1]
```diff
@@ -89,16 +90,16 @@ select
   ticket_comments.count_internal_comments as total_agent_replies,
   
   case when ticket_enriched.is_requester_active = true and ticket_enriched.requester_last_login_at is not null
-    then (({{ dbt_utils.datediff("ticket_enriched.requester_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then({{ dbt_utils.datediff("ticket_enriched.requester_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:93]:
```suggestion
    then ({{ dbt_utils.datediff("ticket_enriched.requester_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
not sure if that's needed but 🤷

File: models/zendesk__ticket_metrics.sql [hunk 2]
```diff
@@ -89,16 +90,16 @@ select
...
       end as requester_last_login_age_minutes,
   case when ticket_enriched.is_assignee_active = true and ticket_enriched.assignee_last_login_at is not null
-    then (({{ dbt_utils.datediff("ticket_enriched.assignee_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then({{ dbt_utils.datediff("ticket_enriched.assignee_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:96]:
```suggestion
    then ({{ dbt_utils.datediff("ticket_enriched.assignee_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
```

File: models/zendesk__ticket_metrics.sql [hunk 3]
```diff
@@ -89,16 +90,16 @@ select
...
       end as assignee_last_login_age_minutes,
   case when lower(ticket_enriched.status) not in ('solved','closed')
-    then (({{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then({{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:99]:
```suggestion
    then( {{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60)
```

File: models/zendesk__ticket_metrics.sql [hunk 4]
```diff
@@ -89,16 +90,16 @@ select
...
       end as unsolved_ticket_age_minutes,
   case when lower(ticket_enriched.status) not in ('solved','closed')
-    then (({{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then({{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:102]:
```suggestion
    then( {{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60)
```

**Review by fivetran-jamie — COMMENTED** [review]
looks good to me! suggested some edits but not sure if they're necessary

**Review by matthew-skinner — APPROVED** [review]

**Review by matthew-skinner — APPROVED** [review]

File: models/zendesk__ticket_metrics.sql (hunk 1 above)
Comment (fivetran-joemarkiewicz) [reply models/zendesk__ticket_metrics.sql:93]: I much prefer having our code be clean and consistent, so I love this and will commit the suggestions. Thanks!! 💯

File: models/zendesk__ticket_metrics.sql [hunk 5]
```diff
@@ -89,16 +90,16 @@ select
...
-    then (({{ dbt_utils.datediff("ticket_enriched.requester_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then ({{ dbt_utils.datediff("ticket_enriched.requester_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
       end as requester_last_login_age_minutes,
   case when ticket_enriched.is_assignee_active = true and ticket_enriched.assignee_last_login_at is not null
-    then (({{ dbt_utils.datediff("ticket_enriched.assignee_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then ({{ dbt_utils.datediff("ticket_enriched.assignee_last_login_at", dbt_utils.current_timestamp(), 'second') }} /60)
       end as assignee_last_login_age_minutes,
   case when lower(ticket_enriched.status) not in ('solved','closed')
-    then (({{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then( {{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:99]:
```suggestion
    then ({{ dbt_utils.datediff("ticket_enriched.created_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
realized i suggested the wrong spacing for a couple 😬

File: models/zendesk__ticket_metrics.sql [hunk 6]
```diff
@@ -89,16 +90,16 @@ select
...
       end as unsolved_ticket_age_minutes,
   case when lower(ticket_enriched.status) not in ('solved','closed')
-    then (({{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60))
+    then( {{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60)
```
Comment (fivetran-jamie) [inline models/zendesk__ticket_metrics.sql:102]:
```suggestion
    then ({{ dbt_utils.datediff("ticket_enriched.updated_at", dbt_utils.current_timestamp(), 'second') }} /60)
```

**Review by fivetran-jamie — APPROVED** [review]
realized i made bad/poorly spaced suggestions for 2 of the case statements -- other than that looks good and i shall approve

**Pull request merged**